common/secret_config.go

```diff
@@ -0,0 +1,20 @@
+package common
+
+import (
+	"fmt"
+)
+
+// SecretConfigV2 contains sensitive config data that must be protected from leakage
+type SecretConfigV2 struct {
+	// SignerPaymentKey is the hex representation of the private payment key, that pays for payload dispersal
+	SignerPaymentKey string
+}
+
+// Check checks config invariants, and returns an error if there is a problem with the config struct
+func (s *SecretConfigV2) Check() error {
+	if s.SignerPaymentKey == "" {
```

Comment:
We can also check the length to verify it's 32 bytes

Reply:
Since the key comes in as a string, which may or may not include `0x` prefix, this would entail a rather awkward check `length must equal 64 or 66`

I think it's reasonable to just let the initialization of the signer fail if the string representing the private key is invalid.